ext/ddtrace.c

```diff
@@ -1096,6 +1096,10 @@ static zval *ddtrace_root_span_data_write(zend_object *object, zend_string *memb
     } else if (zend_string_equals_literal(prop_name, "service")) {
         if (ddtrace_span_is_entrypoint_root(&span->span) && !zend_is_identical(&span->property_service, value)) {
             root_span_data_changed = true;
+            // As per unified service tagging spec if a span is created with a service name different from the global
```

Comment:
Nearly, it needs to be changed when the span is _not_ the root span.
And misses `zval_ptr_dtor(&span->property_version);` to first free to old value.